Popup/Popup.js

```diff
@@ -653,36 +403,275 @@ class Popup extends Component {
+				spotPopupContent();
+			} else if (prevOpen) {
+				forwardHide(null, allComponentProps);
+				spotActivator();
+			}
+		}
 
-		delete rest.no5WayClose;
-		delete rest.onClose;
+		checkScrimNone(allComponentProps);
+	}, [allComponentProps, noAnimation, open, popupOpen, prevOpen, spotActivator, spotPopupContent]);
 
-		return (
-			<FloatingLayer
-				noAutoDismiss={noAutoDismiss}
-				open={this.state.floatLayerOpen}
-				onOpen={this.handleFloatingLayerOpen}
-				onDismiss={this.handleDismiss}
-				scrimType={scrimType}
-			>
-				<SkinnedPopupBase
-					{...rest}
-					data-webos-voice-exclusive
-					onHide={this.handlePopupHide}
-					onShow={this.handlePopupShow}
-					open={this.state.popupOpen >= OpenState.OPENING}
-					spotlightId={this.state.containerId}
-				/>
-			</FloatingLayer>
-		);
-	}
-}
+	const handleDismiss = useCallback((ev) => {
+		forwardCustom('onClose', () => ({detail: ev?.detail}))(null, allComponentProps);
+	}, [allComponentProps]);
+
+	const handleFloatingLayerOpen = useCallback(() => {
+		if (!noAnimation && popupOpen !== OpenState.OPEN) {
+			setPopupOpen(OpenState.OPENING);
+		} else if (popupOpen === OpenState.OPEN && open) {
+			spotPopupContent();
+		}
+	}, [noAnimation, open, popupOpen, spotPopupContent]);
+
+	const handlePopupHide = useCallback((ev) => {
+		forwardHide(ev, allComponentProps);
+
+		setFloatLayerOpen(false);
+		setActivator(null);
+
+		if (!ev.currentTarget || ev.currentTarget.getAttribute('data-spotlight-id') === containerId.current) {
+			spotActivator();
+		}
+	}, [allComponentProps, spotActivator]);
+
+	const handlePopupShow = useCallback((ev) => {
+		forwardShow(ev, allComponentProps);
+
+		setPopupOpen(OpenState.OPEN);
+
+		if (!ev.currentTarget || ev.currentTarget.getAttribute('data-spotlight-id') === containerId.current) {
+			spotPopupContent();
+		}
+	}, [allComponentProps, spotPopupContent]);
+
+	// Spot the content after it's mounted.
+	useEffect(() => {
+		if (open) {
+			// If the popup is open on mount, we need to pause spotlight so nothing steals focus
+			// while the popup is rendering.
+			paused.current.pause();
+			if (getContainerNode(containerId.current)) {
+				spotPopupContent();
+			}
+		}
+
+		const idRef = containerId.current;
+		const keyDownRef = handleKeyDownRef.current;
+
+		return () => {
+			if (open) {
+				off('keydown', keyDownRef);
+			}
+			Spotlight.remove(idRef);
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	useEffect(() => {
+		getDerivedStateFromProps();
+		handleComponentUpdate();
+	}, [getDerivedStateFromProps, handleComponentUpdate]);
+
+	return (
+		<FloatingLayer
+			noAutoDismiss={noAutoDismiss}
+			open={floatLayerOpen}
+			onOpen={handleFloatingLayerOpen}
+			onDismiss={handleDismiss}
+			scrimType={scrimType}
+		>
+			<SkinnedPopupBase
+				{...rest}
+				data-webos-voice-exclusive
+				onHide={handlePopupHide}
+				onShow={handlePopupShow}
+				open={popupOpen >= OpenState.OPENING && open}
```

Comment:
why was the addional condition needed? `&& open`

Reply:
I checked again and the additional condition was unnecessary so I removed it